app/jobs/rstuf/check_job.rb

```diff
@@ -0,0 +1,21 @@
+class Rstuf::CheckJob < Rstuf::ApplicationJob
+  RetryException = Class.new(StandardError)
+  FailureException = Class.new(StandardError)
+  retry_on RetryException, wait: :exponentially_longer, attempts: 10
+
+  queue_with_priority PRIORITIES.fetch(:push)
+
+  def perform(task_id)
+    case status = Rstuf::Client.task_status(task_id)
+    when "SUCCESS"
+      # no-op, all good
```

Comment:
Hi @simi eager to engage in a discussion with you as it will help to improve it.

When the task state is marked as `SUCCESS`, it's essential to verify the corresponding "Task Result Status." Before proceeding, I want to express my apologies; RSTUF requires clarification, and I take responsibility for any confusion—I'll be submitting a PR to address this.

The `SUCCESS` state signifies that the RSTUF executed the task without encountering any issues. However, it's important to note that even in a `SUCCESS` state, the task can still have a result `status` of type bool. We are currently addressing various issues to enhance the result status, but the result `status` will not change.

Specifically for RubyGems, let's examine the task result `status`. If it turns out to be false, we need to handle it by initiating a retry, followed by raising a FailureException with the message "RSTUF task result failed."

Reply:
Clear, I'll update.

Reply:
Btw. wouldn't be better to expose this in unified way? It seems currently this returns directly state of underlying Celery task. But there is no need for this, RubyGems.org do care about state of successful adding artifact to TUF repository.

Reply:
It seems we have 4 outcomes:

1. Pending
2. Completed successfully
3. Completed unsuccessfully (SUCCESS status:false)
4. Failed

Should the client behave differently  between 3 and 4?

Reply:
> Should the client behave differently between 3 and 4?

@martinemde IMHO not, but also I think RSTUF API can be simpler and provide 1 state for 3 and 4. I'll talk to RSTUF next week to ask for change in API. Maybe it is still possible to change.

Reply:
It seems like there's the equivalent of 4XX vs 5XX. You made a bad request vs we failed at running your good request.

The difference could be "failed and you should retry" vs "failed and you should not retry."

Reply:
this was resolved at https://github.com/repository-service-tuf/repository-service-tuf-api/pull/518